File: lib/viper/templates/AppDependencies.m
//
//  <%= @project %>AppDependencies.m
//  <%= @project.downcase %>
//
//  Created by <%= @author %> on <%= @date %>.
//
//

#import "<%= @project %>AppDependencies.h"

#import "RootWireframe.h"


@interface <%= @project %>AppDependencies ()

@end

@implementation <%= @project %>AppDependencies

- (id)initWithWindow:(UIWindow *)window
{
    if ((self = [super init]))
    {
        [self configureDependencies:window];
    }

    return self;
}

- (void)installRootViewController
{
    // *** present first wireframe here
}

- (void)configureDependencies:(UIWindow *)window
{
    // -----
    // root classes
    RootWireframe *rootWireframe = [[RootWireframe alloc] initWithWindow:window];
    // *** add datastore

    // *** module initialization
}


@end

File: lib/boa/templates/swift/AppDependencies.swift
//
//  <%= @project %>AppDependencies.swift
//  <%= @project %>
//
//  Created by <%= @author %> on <%= @date %>.
//
//

import Foundation
import UIKit

class <%= @project %>AppDependencies: NSObject
{

    class func initWithWindow(_ window: UIWindow) -> <%= @project %>AppDependencies
    {

        let obj = <%= @project %>AppDependencies()
        obj.configureDependencies(window)

        return obj
    }

    func installRootViewController()
    {
        // *** present first wireframe here
    }

    func configureDependencies(_ window: UIWindow)
    {
        // -----
        // root classes
        let rootWireframe = RootWireframe.init(window: window)
        // *** add datastore

        // *** module initialization
    }
}
